alacritty/src/display/damage.rs

```diff
@@ -0,0 +1,345 @@
+use std::cmp;
+
+use glutin::Rect;
+
+use alacritty_terminal::selection::SelectionRange;
+use alacritty_terminal::term::{DamagedLine, RenderableCursor, SizeInfo, TermDamage};
+use alacritty_terminal::vi_mode::ViModeCursor;
+
+/// Minimum part of damaged square when intersecting damaged rectangles.
+pub const MIN_DAMAGED_SQUARE_PART: f32 = 0.4;
+
+/// An error when we for look ahead in iterator, used to try to collect rectangles which pretty
+/// close to each other, but don't fit into `MIN_DAMAGED_SQUARE_PART` just by looking on the first
+/// next damaged rectangle.
+pub const MIN_DAMAGED_SQUARE_PART_ERR: f32 = MIN_DAMAGED_SQUARE_PART / 2.;
+
+pub struct RendererDamage<'a> {
```

Comment:
Small comment what this struct is used for might be nice.

This is all damage information required by the renderer for one frame, right?

Reply:
It's all damage information collected from alacritty_terminal.